android/src/main/java/com/optimizely/optimizely_flutter_sdk/helper_classes/Utils.java

```diff
@@ -72,7 +72,7 @@ public static List<OptimizelyDecideOption> getDecideOptions(List<String> options
     }
 
     public static List<ODPSegmentOption> getSegmentOptions(List<String> options) {
-        if(options == null || options.isEmpty()) {
+        if(options == null) {
```

Comment:
Flutter sends empty options list to native sdk's in case no options are provided. The removed check causes the options to be null for empty list (which is default flutter behaviour) which is not supported by native android-sdk.

Reply:
@muzahidul-opti can we add a unit test that verifies that in case no options are provided, flutter sends empty list to native sdk's? and then add an integration test to @optimizely-flutter-testapp to validate the response.